AssetTracking/AssetTracking/Models/Sites.cs

```diff
@@ -0,0 +1,166 @@
+                            .Groups[groupsId]
+                            .Sites[siteId]
+                            .Lists[listId]
+                            .Request().Expand("fields")
+                            .GetAsync();
+            return await list;
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="graphServiceClient"></param>
+        /// <param name="listItem"></param>
+        /// <param name="groupId"></param>
+        /// <param name="siteId"></param>
+        /// <param name="listId"></param>
+        /// <returns></returns>
+        public static async Task<ListItem> AddListItem(GraphServiceClient graphServiceClient, ListItem listItem, string groupId, string siteId, string listId)
+        {
+            ListItem result = new ListItem();
+            try
+            {
+                result = await graphServiceClient.Groups[groupId]
+                            .Sites[siteId]
+                            .Lists[listId]
+                              .Items
+                              .Request()
+                              .AddAsync(listItem);
+            }
+            catch (Exception ex)
+            {
+      
+            }
+
+            return result;
+        }
+
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="graphServiceClient"></param>
+        /// <param name="fieldValueSet"></param>
+        /// <param name="groupId"></param>
+        /// <param name="siteId"></param>
+        /// <param name="listId"></param>
+        /// <param name="listItemId"></param>
+        /// <returns></returns>
+        public static async Task<FieldValueSet> UpdateListItem(GraphServiceClient graphServiceClient, FieldValueSet fieldValueSet, string groupId, string siteId, string listId, string listItemId)
+        {
+            FieldValueSet result = new FieldValueSet();
+            try
+            {
+                result = await graphServiceClient.Groups[groupId]
+                           .Sites[siteId]
+                           .Lists[listId]
+                              .Items[listItemId]
+                              .Fields
+                              .Request()
+                              .UpdateAsync(fieldValueSet);
+
+                return result;
+            }
+            catch (Exception ex)
+            {
+       
+            }
+            return result;
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="graphServiceClient"></param>
+        /// <param name="listItem"></param>
+        /// <param name="groupId"></param>
+        /// <param name="siteId"></param>
+        /// <param name="listId"></param>
+        /// <returns></returns>
+        public static async Task<ListItem> DeleteListItem(GraphServiceClient graphServiceClient, ListItem listItem, string groupId, string siteId, string listId)
+        {
+            throw new NotImplementedException();
```

Comment:
Provide the implementation for this method

Reply:
This is still a working progress